### PR TITLE
Remove tasks that were moved to eo-learn-examples

### PR DESCRIPTION
**Pull request by zigaLuksic**

### CODE REVIEW

File: eolearn/features/bands_extraction.py
Comment (mlubej) [inline eolearn/features/bands_extraction.py:1]: rename file, perhaps to `ndi.py`

File: eolearn/io/geometry_io.py
Comment (mlubej) [inline eolearn/io/geometry_io.py:1]: The only remaining task now is the `VectorImportTask`, so perhaps there is no more need to have `_BaseVectorImportTask`

Comment (mlubej) [inline eolearn/io/geometry_io.py:1]: Additionally, I would remove the `SentinelHubSen2corTask` from `eolearn/io/sentinelhub_process.py`

File: eolearn/mask/extra/cloud_mask.py
Comment (mlubej) [inline eolearn/mask/extra/cloud_mask.py:1]: the `ClassifierType` can be removed

File: eolearn/ml_tools/__init__.py
Comment (mlubej) [inline eolearn/ml_tools/__init__.py:1]: There are also some deprecated tools in the `ml_tools/utils.py`, should we remove them? Potentially also other deprecated things? or perhaps it's better to do this in a dedicated PR

**Comment by mlubej** [comment]
I also noticed that the `example_data/REFERENCE_SCENES.npy` isn't used anywhere now, since it was moved to `eo-learn-examples/extra-tasks/radiometric_normalization/_test_data.npy`

File: eolearn/io/geometry_io.py
Comment (zigaLuksic) [reply eolearn/io/geometry_io.py:1]: did we move it? if so, i will

Comment (mlubej) [reply eolearn/io/geometry_io.py:1]: we didn't move it, that was my comment in https://github.com/sentinel-hub/eo-learn-examples/pull/34#issue-1880530818

> There is one more task which I have not ported, the SentinelHubSen2corTask. I think it pulls in too much code for a very simple thing, which is now easier achievable with the new code, so IMHO the SentinelHubSen2corTask has become redundant and I would just deprecated/remove it in eo-learn.

Comment (mlubej) [reply eolearn/io/geometry_io.py:1]: feel free to do it separately, perhaps the review will be clearer

File: eolearn/mask/extra/cloud_mask.py
Comment (zigaLuksic) [reply eolearn/mask/extra/cloud_mask.py:1]: done

**Review by mlubej — APPROVED** [review]
